src/mono/mono/mini/mini-amd64.c

```diff
@@ -2174,6 +2174,14 @@ mono_arch_get_llvm_call_info (MonoCompile *cfg, MonoMethodSignature *sig)
 				return linfo;
 			}
 
+			if ((t->type == MONO_TYPE_GENERICINST) && !cfg->full_aot) {
+				MonoClass *klass = mono_class_from_mono_type_internal (t);
```

Comment:
This should also check !sig->pinvoke, since calls to native code might not use this calling convention.

Reply:
@fanyang-mono @tannergooding I think we should do this in two stages.  First as disallow this kind of argument passing at interop boundaries, and as followup follow the ABI/calling convention on pinvokes and reverse pinvokes.

@vargaz what do we need to check to block this on reverse pinvokes?  `sig->call_convention == MONO_CALL_DEFAULT` ?

Reply:
sig->pinvoke is TRUE in both cases.